io.openems.edge.timeofusetariff.entsoe/src/io/openems/edge/timeofusetariff/entsoe/TouEntsoeImpl.java

```diff
@@ -69,82 +66,75 @@ public TouEntsoeImpl() {
 		);
 	}
 
+	private final Consumer<Value<Integer>> updateCurrency = currency -> {
+		this.currency = currency.asEnum();
```

Comment:
I don't think it is necessary to keep a local 'currency', because we can always read it from Meta.